src/scenes/NftDetailPage/NftDetailPage.tsx

```diff
@@ -27,12 +31,21 @@ type Props = {
 function NftDetailPage({ nftId }: Props) {
   const { query } = useRouter();
 
+  const screenWidth = useBreakpoint();
+  const isMobileScreen = screenWidth === size.mobileLarge || screenWidth === size.mobile;
+
   const username = window.location.pathname.split('/')[1];
-  const collectionId = query.collectionId;
+  const collectionId = query.collectionId as string;
   // TODO: Should refactor to utilize navigation context instead of session storage
   const isFromCollectionPage =
     globalThis?.sessionStorage?.getItem('prevPage') === `/${username}/${collectionId}`;
 
+  const collectionNfts = useCollectionById({ id: collectionId })?.nfts ?? [];
+  const collectionIndexOfNft = collectionNfts?.findIndex((nft) => nft.id === nftId) ?? 0;
```

Comment:
i think we should `useMemo` here since `findIndex` takes O(n)

Reply:
yupp that's the perfect way to memo!

you could also use optional chaining before square brackets as well. here's an example (renamed `collectinIndexOfNft` => `nftIndex`):
```js
const prevNftId = collectionNfts?.[nftIndex - 1]?.id ?? null
const nextNftId = collectionNfts?.[nftIndex + 1]?.id ?? null
```

you could also do something more exhaustive like the following, esp. since it would be a red flag for an NFT to not belong to a collection:
```js
const collectionNfts = useCollectionById({ id: collectionId })

const { prevNftId, nextNftId } = useMemo(() => {
  if (!collectionNfts) {
    // TODO: send error to sentry. technically all NFTs should belong to a collection.
    return {
      prevNftId: null,
      nextNftId: null,
    }
  }

  const nftIndex = collectionNfts.findIndex((nft) => nft.id === nftId)
  if (nftIndex === -1) {
    // TODO: send error to sentry. technically the NFT should exist within the collection.
    return {
      prevNftId: null,
      nextNftId: null,
    }
  }
  return {
    prevNftId: collectionNfts[nftIndex - 1]?.id ?? null,
    nextNftId: collectionNfts[nftIndex + 1]?.id ?? null,
  }
}, [collectionNfts])
```